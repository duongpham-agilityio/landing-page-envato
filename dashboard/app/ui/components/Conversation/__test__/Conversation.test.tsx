import { Conversation } from '../..';

const CONVERSATION_PROPS = {
  messages: [
    {
      date: {
        nanoseconds: 1,
        seconds: 11,
      },
      id: '1',
      senderId: '2',
      text: 'Hello',
    },
  ],
};

const mockMobileMediaQuery = () =>
  Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation(() => ({
      matches: true,
      media: '480px',
      onchange: null,
      addListener: jest.fn(),
      removeListener: jest.fn(),
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    })),
  });

describe('ChatMember component', () => {
  mockMobileMediaQuery();
  test('Should render match with snapshot.', () => {
    const { container } = render(<Conversation {...CONVERSATION_PROPS} />);

    expect(container).toMatchSnapshot();
  });
});
