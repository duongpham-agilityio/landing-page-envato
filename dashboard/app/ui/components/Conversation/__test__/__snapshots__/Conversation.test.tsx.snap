// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatMember component Should render match with snapshot. 1`] = `
<div>
  <div
    class="css-j0gh2o"
  >
    <div
      class="css-t4f5wq"
    >
      <h3
        class="chakra-heading css-1rl4kbt"
      >
        <div
          class="css-1wloztz"
        >
          <div
            class="css-1826kcq"
          >
            <div
              class="css-1lg10hu"
            >
              <span
                class="chakra-avatar css-1r3pczz"
              >
                <svg
                  aria-label=" avatar"
                  class="chakra-avatar__svg css-16ite8i"
                  role="img"
                  viewBox="0 0 128 128"
                >
                  <path
                    d="M103,102.1388 C93.094,111.92 79.3504,118 64.1638,118 C48.8056,118 34.9294,111.768 25,101.7892 L25,95.2 C25,86.8096 31.981,80 40.6,80 L87.4,80 C96.019,80 103,86.8096 103,95.2 L103,102.1388 Z"
                    fill="currentColor"
                  />
                  <path
                    d="M63.9961647,24 C51.2938136,24 41,34.2938136 41,46.9961647 C41,59.7061864 51.2938136,70 63.9961647,70 C76.6985159,70 87,59.7061864 87,46.9961647 C87,34.2938136 76.6985159,24 63.9961647,24"
                    fill="currentColor"
                  />
                </svg>
                <div
                  class="chakra-avatar__badge css-1wcnhnd"
                />
              </span>
            </div>
            <div
              class="css-dvxtzn"
            >
              <p
                class="chakra-text css-16h4vrm"
              />
            </div>
          </div>
        </div>
      </h3>
    </div>
    <div
      class="css-h5aeco"
    >
      <div
        class="css-1guw43s"
      >
        <div
          class="css-14cwlhy"
        >
          <span
            class="chakra-avatar css-1cw6pp6"
          >
            <svg
              aria-label=" avatar"
              class="chakra-avatar__svg css-16ite8i"
              role="img"
              viewBox="0 0 128 128"
            >
              <path
                d="M103,102.1388 C93.094,111.92 79.3504,118 64.1638,118 C48.8056,118 34.9294,111.768 25,101.7892 L25,95.2 C25,86.8096 31.981,80 40.6,80 L87.4,80 C96.019,80 103,86.8096 103,95.2 L103,102.1388 Z"
                fill="currentColor"
              />
              <path
                d="M63.9961647,24 C51.2938136,24 41,34.2938136 41,46.9961647 C41,59.7061864 51.2938136,70 63.9961647,70 C76.6985159,70 87,59.7061864 87,46.9961647 C87,34.2938136 76.6985159,24 63.9961647,24"
                fill="currentColor"
              />
            </svg>
          </span>
          <div
            class="css-u4p24i"
          >
            <div
              class="css-1sl4v5d"
              data-testid="image-container"
            >
              Hello
            </div>
            <div
              class="css-17xejub"
            />
            <p
              class="chakra-text css-t1z24v"
            >
              08:00 AM
            </p>
          </div>
        </div>
      </div>
      <form
        class="chakra-stack css-tl3ftk"
      >
        <div
          class="css-1nlrkd1"
        >
          <div
            class="css-u4p24i"
          >
            <div
              class="css-14ethi5"
            />
          </div>
          <p
            class="chakra-text css-rj1t1b"
          />
          <button
            class="chakra-button css-qcuulc"
            disabled=""
            type="submit"
          >
            <span
              class="chakra-button__icon css-1wh2kri"
            >
              <svg
                fill="none"
                height="20"
                viewBox="0 0 20 20"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M15.0586 7.09154L7.92522 3.52487C3.13355 1.12487 1.16689 3.09153 3.56689 7.8832L4.29189 9.3332C4.50022 9.7582 4.50022 10.2499 4.29189 10.6749L3.56689 12.1165C1.16689 16.9082 3.12522 18.8749 7.92522 16.4749L15.0586 12.9082C18.2586 11.3082 18.2586 8.69153 15.0586 7.09154ZM12.3669 10.6249H7.86689C7.52522 10.6249 7.24189 10.3415 7.24189 9.99987C7.24189 9.6582 7.52522 9.37487 7.86689 9.37487H12.3669C12.7086 9.37487 12.9919 9.6582 12.9919 9.99987C12.9919 10.3415 12.7086 10.6249 12.3669 10.6249Z"
                  fill="white"
                />
              </svg>
            </span>
            Send
          </button>
        </div>
      </form>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
